### PR TITLE
Add setregs shellcode helper for i386 to demonstrate regsort

### LINKED ISSUE #441: shellcraft.{i386,amd64}.linux.syscall don't check register ordering
If you use a destination register as a source register, things break.

In this example, the call ends up being `write(edi, edi, eax)` since `ebx` is clobbered in the first instruction.

```
shellcraft i386.linux.syscall SYS_write edi ebx eax -f asm
    /* call write('edi', 'ebx', 'eax') */
    mov ebx, edi
    mov ecx, ebx
    mov edx, eax
    push 0x4
    pop eax
    int 0x80
```

### PR DESCRIPTION
**Pull request by zachriggle**
Fixes #441.

```py
>>> print shellcraft.setregs({'eax':'ebx','ebx':3,'edx':'eax','esi':'edi','edi':'esi','ebp':'ebx'})
    mov edx, eax
    mov eax, ebx
    mov ebp, ebx
    push 0x3
    pop ebx
    xchg edi, esi
```

### CODE REVIEW

**Comment by zachriggle** [comment]
Well, *can* fix #441 if you integrate it into `i386.mov`. However, then all the unit tests have to be re-written as the register ordering changes.

File: pwnlib/shellcraft/templates/i386/setregs.asm [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  from pwnlib.shellcraft.i386 import mov
+%>
+<%page args="reg_context, stack_allowed = True"/>
+<%docstring>
+Sets multiple registers, taking any register dependencies into account
+(i.e., given eax=1,ebx=eax, set ebx first).
+
+Args:
+    reg_context (dict): Desired register context
+    stack_allowed (bool): Can the stack be used?
```
Comment (kokjo) [inline pwnlib/shellcraft/templates/i386/setregs.asm]: stack_allowed is unused. Why?

Comment (zachriggle) [inline pwnlib/shellcraft/templates/i386/setregs.asm]: Oversight :)

**Comment by br0ns** [comment]
Looks good.  Merging.

**Pull request merged**